revscoring/scorers/scorer.py

```diff
@@ -1,141 +1,320 @@
                 `Feature` s provided to the constructor
-                
+        
         :Returns:
-            A dictionary of test results.
+            A dictionary of model statistics.
         """
         raise NotImplementedError()
+        
     
-    
-    def score(self, values, **opts):
+    def test(self, values_labels):
         """
-        Make a prediction or otherwise use the model to generate a score.
+        Tests the model against a labeled data.  Note that test data should be
+        withheld from from train data.
         
         :Parameters:
-            values_scores : `iterable`((`<feature_values>`, `<score>`))
-                an iterable of labeled data Where <values_scores> is an ordered
+            values_labels : `iterable`((`<feature_values>`, `<label>`))
+                an iterable of labeled data Where <values_labels> is an ordered
                 collection of predictive values that correspond to the
                 `Feature` s provided to the constructor
-            opts : dict
-                optional arguments to affect how scores are generated/returned
                 
         :Returns:
-            A dictionary of score values
+            A dictionary of test results.
         """
         raise NotImplementedError()
     
     
-    def _validate_features(self, values):
-        """
-        Checks the features against provided values to confirm types,
-        ordinality, etc.
-        """
-        return [feature.return_type(value)
-                for feature, value in zip(self.feature, values)]
-            
+     
     @classmethod
     def load(cls, f):
         """
-        Reads serialized model information from a file.
+        Reads serialized model information from a file.  Make sure to open
+        the file as a binary stream.
         """
         return pickle.load(f)
-    
+
     def dump(self, f):
         """
-        Writes serialized model information to a file.
+        Writes serialized model information to a file.  Make sure to open the
+        file as a binary stream.
         """
         pickle.dump(self, f)
+
+    @classmethod
+    def from_config(self, config, key):
+        """
+        Constructs a model from configuration.
+        """
+        section = config['models'][key]
+        return cls.load(open(section['file'], 'rb'))
+        
+
+
+class ScikitLearnClassifier(MLScorerModel):
+    
+    def __init__(self, features, classifier_model, language=None):
+        super().__init__(features, language=language)
+        self.classifier_model = classifier_model
+    
+    def train(self, values_labels):
+        """
+        
+        :Returns:
+            A dictionary with the fields:
+            
+            * seconds_elapsed -- Time in seconds spent fitting the model
+        """
+        start = time.time()
+        
+        values, labels = zip(*values_labels)
+        
+        # Fit SVC model
```

Comment:
This comment looks outdated: `classifier_model` might not be an SVC model, right?